CRAWLING.md

```diff
@@ -0,0 +1,177 @@
+
+   * (Optional) Check if there are contribution guidelines and add the file name with the key `guidelines` (e.g. `CONTRIBUTING.md`). If specified, the *Contribute* button will link directly to the file instead of the repository root.
+
+   * (Optional) Query GitHub [topics](https://docs.github.com/en/rest/reference/repos#get-all-repository-topics) and add the array of topics with the key `topics` for each repo to allow searching projects by topic and displaying them on the detail popup.
+
+   * (Optional) Calculate the [Repository Activity Score](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/master/patterns/2-structured/repository-activity-score.md) to define a meaningful order for the projects. Sort entries by score descending. Add it with the key `score`
+
+3. Write the resulting list of projects with all metadata to the file ```repos.json``` to serve all projects in the portal.
+
+With this approach, projects can self-register to the portal by adding the ```inner-source``` topic to their repository and specifying additional metadata inside an ```innersource.json``` file.
+
+For more information about this file, see [Listing Project in the Project Portal for InnerSource](CONTRIBUTING.md#listing-project-in-the-project-portal-for-innersource)
+
+
+## Extra project metadata via `innersource.json`
+
+To provide more details or add a custom logo, add an `innersource.json` file in the root of your repository with the following format:
+
+``` json
+{
+ "title": "Readable Project Name (optional)",
+ "motivation": "A short statement why this project is InnerSource and why contributors should care (optional)",
+ "contributions": [
+   "List",
+   "Of",
+   "Requested",
+   "Contribtions",
+   "Like",
+   "Bugfixes",
+   "Features",
+   "Infrastructure",
+   "Documentation",
+   "..."
+ ],
+ "skills": [
+   "Skills",
+   "Required",
+   "To",
+   "Contribute",
+   "Like",
+   "Node.js",
+   "Java",
+   "C++",
+   "..."
+ ],
+ "logo": "path/to/your/project-logo.png (optional)",
+ "docs": "http://url/to/project/documentation (optional)",
+ "language": "JavaScript (optional)"
+}
+```
+
+> *Note:* The property `language` overrides GitHub's [programming language detection](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-repository-languages). Supported values are all entries of type `programming` from the [linguist language list](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml).
+
+
+## Reference Implementations
+
+*NOTE*: This section needs to be expanded
+
+You will likely have to adapt all of these crawler implementations to your exact setup. However they may still get you off the ground faster than starting from scratch.
+
+* with jq: <TBD SAMPLE HERE>
+* with ruby: https://github.com/spier/innersource-crawler-ruby
+* with python: https://github.com/zkoppert/innersource-crawler
```

Comment:
🎉

Reply:
Oh wow, what dark GitHub magic did you apply @zkoppert to find this so quickly?
Alerts on anything with `https://github.com/zkoppert` or what? :)

Reply:
haha. Nope, I just am subscribed to any changes in this repo. I did that to help me keep track of any patches I need to apply to my deployments of this project.

Reply:
Well that actually is a bit more logical than my idea. 
Although I kind of liked the idea of GitHub staff having secret super-power notifications :)

Reply:
haha yeah that would be wonderful!